### PR TITLE
Attlasian iam

### PR DESCRIPTION
**Pull request by Shellyber**
## Status
- [ ] In Progress
- [x] Ready
- [ ] In Hold - (Reason for hold)

## Related Issues
fixes: https://github.com/demisto/etc/issues/32451

## Minimum version of Demisto
- [ ] 5.0.0
- [ ] 5.5.0
- [x] 6.0.0

## Does it break backward compatibility?
   - [x] No

## Must have
- [x] Tests
- [x] Documentation

### CODE REVIEW

File: Packs/ApiModules/Scripts/IAMApiModule/IAMApiModule.py [hunk 1]
```diff
@@ -306,18 +306,18 @@ def __init__(self, is_create_enabled=True, is_disable_enabled=True, is_update_en
...
         :param client: (Client) The integration Client object that implements a get_user() method
         :param args: (dict) The `iam-get-user` command arguments
+        :param identifier: (str) The attribute name to search by.
         :return: (IAMUserProfile) The user profile object.
         """
         user_profile = IAMUserProfile(user_profile=args.get('user-profile'))
         try:
-            email = user_profile.get_attribute('email')
-            user_app_data = client.get_user(email)
+            identifier = user_profile.get_attribute(identifier)
```
Comment (dantavori) [inline Packs/ApiModules/Scripts/IAMApiModule/IAMApiModule.py:319]:
```suggestion
            identifier = user_profile.get_attribute(self.attr)
```

File: Packs/ApiModules/Scripts/IAMApiModule/IAMApiModule.py [hunk 2]
```diff
@@ -330,7 +330,7 @@ def get_user(self, client, args):
                     action=IAMActions.GET_USER,
                     active=user_app_data.is_active,
                     iden=user_app_data.id,
-                    email=email,
+                    email=user_profile.get_attribute('email'),
```
Comment (dantavori) [inline Packs/ApiModules/Scripts/IAMApiModule/IAMApiModule.py:333]:
```suggestion
                    email=user_profile.get_attribute(self.attr),
```

File: Packs/Attlasian/Integrations/Attlasian_IAM/Attlasian_IAM.py [hunk 3]
```diff
@@ -0,0 +1,230 @@
...
+
+    mapper_in = params.get('mapper_in')
+    mapper_out = params.get('mapper_out')
+    is_create_enabled = params.get("create_user_enabled")
+    is_disable_enabled = params.get("disable_user_enabled")
+    is_update_enabled = demisto.params().get("update_user_enabled")
+    create_if_not_exists = demisto.params().get("create_if_not_exists")
+
+    iam_command = IAMCommand(is_create_enabled, is_disable_enabled, is_update_enabled,
+                             create_if_not_exists, mapper_in, mapper_out)
```
Comment (dantavori) [inline Packs/Attlasian/Integrations/Attlasian_IAM/Attlasian_IAM.py:180]:
```suggestion
    iam_command = IAMCommand(is_create_enabled, is_disable_enabled, is_update_enabled,
                             create_if_not_exists, mapper_in, mapper_out, attr='username')
```

File: Packs/Attlasian/Integrations/Attlasian_IAM/Attlasian_IAM.py [hunk 4]
```diff
@@ -0,0 +1,230 @@
...
+        verify=verify_certificate,
+        proxy=proxy,
+        headers=headers,
+        ok_codes=(200, 201)
+    )
+
+    demisto.debug(f'Command being called is {command}')
+
+    if command == 'iam-get-user':
+        user_profile = iam_command.get_user(client, args, 'username')
```
Comment (dantavori) [inline Packs/Attlasian/Integrations/Attlasian_IAM/Attlasian_IAM.py:200]:
```suggestion
        user_profile = iam_command.get_user(client, args)
```

File: Packs/Attlasian/pack_metadata.json [hunk 5]
```diff
@@ -0,0 +1,14 @@
+{
+    "name": "Atlassian Cloud IT Admin",
+    "description": "Atlassian Cloud IT Admin Integration consists of set of endpoints that allows the customers to do the CRUD operation.",
```
Comment (dantavori) [inline Packs/Attlasian/pack_metadata.json:3]:
```suggestion
    "name": "Atlassian IAM",
    "description": "Atlassian IAM Integration consists of set of endpoints that allows the customers to do the CRUD operation.",
```

File: Packs/Attlasian/Integrations/Attlasian_IAM/command_examples [hunk 6]
```diff
@@ -0,0 +1,4 @@
+!iam-disable-user user-profile=`{"email": "testdemisto@paloaltonetworks.com", "username": "Demisto"}` using="Atlassian IAM_instance_1"
+!iam-get-user user-profile=`{"email": "test@paloaltonetworks.com", "username": "testDemisto"}` using="Atlassian IAM_instance_1"
+!iam-create-user user-profile=`{"emails":[{"value":"testatlas@paloaltonetworks.com","type":"work","primary":"true"}],"is_active": "true", "userName":"testatlas@paloaltonetworks.com"}` using="Atlassian IAM_instance_1"
```
Comment (dantavori) [inline Packs/Attlasian/Integrations/Attlasian_IAM/command_examples:3]: this should be the same format as the other commands

**Review by dantavori — CHANGES_REQUESTED** [review]
looks very good! see my comments

File: Packs/Attlasian/pack_metadata.json (hunk 5 above)
Comment (dantavori) [reply Packs/Attlasian/pack_metadata.json:3]: maybe we can think of a better description here (this one is a bit too generic)

File: Packs/ApiModules/Scripts/IAMApiModule/IAMApiModule.py (hunk 2 above)
Comment (Shellyber) [reply Packs/ApiModules/Scripts/IAMApiModule/IAMApiModule.py:333]: Here its ok, we want to return the email anyways to the Human readable entry

File: Packs/ApiModules/Scripts/IAMApiModule/IAMApiModule.py [hunk 7]
```diff
@@ -298,26 +298,28 @@ def __init__(self, is_create_enabled=True, is_disable_enabled=True, is_update_en
         :param create_if_not_exists: (bool) Whether or not to create a user if does not exist in the application
         :param mapper_in: (str) Incoming mapper from the application to Cortex XSOAR
         :param mapper_out: (str) Outgoing mapper from the Cortex XSOAR to the application
+        :param attr: (str) An attribute to pass to get_user() function that identifies with what field to get the user.
```
Comment (dantavori) [inline Packs/ApiModules/Scripts/IAMApiModule/IAMApiModule.py:301]:
```suggestion
        :param attr: (str) The name of the identifier attribute of users in the application.
```

File: Packs/ApiModules/Scripts/IAMApiModule/IAMApiModule.py [hunk 8]
```diff
@@ -298,26 +298,28 @@ def __init__(self, is_create_enabled=True, is_disable_enabled=True, is_update_en
...
         self.mapper_out = mapper_out
+        self.attr = attr
 
     def get_user(self, client, args):
         """ Searches a user in the application and updates the user profile object with the data.
             If not found, the error details will be resulted instead.
-
         :param client: (Client) The integration Client object that implements a get_user() method
         :param args: (dict) The `iam-get-user` command arguments
+        :param identifier: (str) The attribute name to search by.
```
Comment (dantavori) [inline Packs/ApiModules/Scripts/IAMApiModule/IAMApiModule.py:316]: remove this line

**Review by dantavori — APPROVED** [review]

**Pull request merged**